File: src/main/scala/nl/rug/jbi/search/ssap/model/System.scala
package nl.rug.jbi.search.ssap.model

import scala.collection.mutable
import scala.collection.mutable.ArrayBuffer

class System (val patterns: mutable.Buffer[Pattern]) {

  def toXml() = {
    <system>
      {patterns.map(_.toXml())}
    </system>
  }
}

object System {
  def fromXml(node: scala.xml.Node):System = {
    val s:System = new System(ArrayBuffer[Pattern]())
    (node \\ "pattern").map(p => s.patterns+=Pattern.fromXml(p))
    s
  }
}

File: ssap/src/main/java/nl/rug/jbi/search/ssap/model/System.java
package nl.rug.jbi.search.ssap.model;

import com.thoughtworks.xstream.annotations.XStreamAlias;
import com.thoughtworks.xstream.annotations.XStreamImplicit;

import java.util.List;

@XStreamAlias("system")
public class System {
    @XStreamImplicit
    public List<Pattern> patternList;

    private System (List<Pattern> patternList) {
        this.patternList = patternList;
    }
}
